include/utils.hpp

```diff
@@ -395,6 +396,22 @@ static_assert(std::is_same_v<std::vector<int>, type_transform<std::vector, int>>
 static_assert(std::is_same_v<std::tuple<std::vector<int>, std::vector<float>>, type_transform<std::vector, std::tuple<int, float>>>);
 static_assert(std::is_same_v<void, type_transform<std::vector, void>>);
 
+#ifdef __cpp_lib_hardware_interference_size
+static constexpr const std::size_t kCacheLine = std::hardware_destructive_interference_size;
+#else
+static constexpr const std::size_t kCacheLine = 64;
+#endif
+
+constexpr bool
+isPower2(std::unsigned_integral auto value) {
+    return !(value == 0) && !(value & (value - 1));
+}
+
+template<std::unsigned_integral auto N>
+static constexpr bool is_power2_v = isPower2(N);
+
+#define FWD(x) std::forward<decltype(x)>(x)
```

Comment:
While I initially liked/introduce this short-hand... the question is whether this is clean from a STL-lib point-of-view. Maybe it's better to explicitly state the forwarding and types where it is needed rather than using this auto-forward.

Also this may not be obvious to new C++ developers how this works... :thinking:

Reply:
ACK, I have removed the auto-forwarding